core/runtime/jsruntime.rs

```diff
@@ -294,7 +305,10 @@ pub(crate) const NO_OF_BUILTIN_MODULES: usize = 2;
 pub struct JsRuntime {
   pub(crate) inner: InnerIsolateState,
   pub(crate) allocations: IsolateAllocations,
-  extensions: Vec<Extension>,
+  // Contains paths of source files that were executed in
+  // [`JsRuntime::init_extension_js`]. This field is populated only if a
+  // snapshot is being created.
+  files_loaded_from_fs_during_snapshot: Vec<&'static str>,
```

Comment:
Extensions are no longer stored on the `JsRuntime` - they are consumed fully during initialization.